### PR TITLE
Prometheus: Flavor/version configuration (#57554)

### PR DESCRIPTION
**Pull request by gtk-grafana**
* Revert "Revert "Prometheus: Type and flavor configuration (#56496)" (#57552)" This reverts commit 2432ce619a7e62026685df14f60116ea4fb3447d.
* Adds new fields and documentation for Prometheus datasource configuration: prometheus type, and version

(cherry picked from commit f93c3acc51180cee78222f2df95ac7b380d1a616)

**What is this feature?**

[Add a brief description of what the feature or update does.]

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by gguillotte-grafana — APPROVED** [review]

**Comment by gtk-grafana** [comment]
Merge https://github.com/grafana/grafana/pull/57564 before this one!

**Review by gguillotte-grafana — COMMENTED** [review]
Approving docs, pending approval of code changes.

**Pull request merged**
